include/seqan3/std/charconv

```diff
@@ -0,0 +1,1010 @@
+{
+    using tx = itoa::traits<_Tp>;
+    auto diff = last - first;
+
+    if (tx::digits <= diff || tx::width(value) <= diff)
+        return {tx::convert(value, first), {}};
+    else
+        return {last, errc::value_too_large};
+}
+
+template <typename _Tp>
+inline to_chars_result to_chars_integral(char* first, char* last, _Tp value, int base,
+                    true_type)
+{
+    auto x = to_unsigned(value);
+    if (value < 0 && first != last)
+    {
+        *first++ = '-';
+        x = complement(x);
+    }
+
+    return to_chars_integral(first, last, x, base, false_type());
+}
+
+template <typename _Tp>
+inline to_chars_result to_chars_integral(char* first, char* last, _Tp value, int base,
+                    false_type)
+{
+    if (base == 10)
+        return to_chars_itoa(first, last, value, false_type());
+
+    auto p = last;
+    while (p != first)
+    {
+        auto c = value % base;
+        value /= base;
+        *--p = "0123456789abcdefghijklmnopqrstuvwxyz"[c];
+        if (value == 0)
+            break;
+    }
+
+    auto len = last - p;
+    if (value != 0 || !len)
+        return {last, errc::value_too_large};
+    else
+    {
+        memmove(first, p, len);
+        return {first + len, {}};
+    }
+}
+
+template <typename _It, typename _Tp, typename _Fn, typename... _Ts>
+inline from_chars_result sign_combinator(_It first, _It last, _Tp& value, _Fn f, _Ts... args)
+{
+    using tl = numeric_limits<_Tp>;
+    decltype(to_unsigned(value)) x;
+
+    bool neg = (first != last && *first == '-');
+    auto r = f(neg ? first + 1 : first, last, x, args...);
+
+    switch (r.ec)
+    {
+    case errc::invalid_argument:
+        return {first, r.ec};
+    case errc::result_out_of_range:
+        return r;
+    default:
+        break;
+    }
+
+    if (neg)
+    {
+        if (x <= complement(to_unsigned(tl::min())))
+        {
+            x = complement(x);
+            memcpy(&value, &x, sizeof(x));
+            return r;
+        }
+    }
+    else
+    {
+        if (x <= (tl::max)())
+        {
+            value = x;
+            return r;
+        }
+    }
+
+    return {r.ptr, errc::result_out_of_range};
+}
+
+template <typename _Tp>
+inline bool in_pattern(_Tp c)
+{
+    return '0' <= c && c <= '9';
+}
+
+struct in_pattern_result
+{
+    bool ok;
+    int val;
+
+    explicit operator bool() const { return ok; }
+};
+
+template <typename _Tp>
+inline in_pattern_result in_pattern(_Tp c, int base)
+{
+    if (base <= 10)
+        return {'0' <= c && c < '0' + base, c - '0'};
+    else if (in_pattern(c))
+        return {true, c - '0'};
+    else if ('a' <= c && c < 'a' + base - 10)
+        return {true, c - 'a' + 10};
+    else
+        return {'A' <= c && c < 'A' + base - 10, c - 'A' + 10};
+}
+
+template <typename _It, typename _Tp, typename _Fn, typename... _Ts>
+inline from_chars_result subject_seq_combinator(_It first, _It last, _Tp& value, _Fn f, _Ts... args)
```

Comment:
in theory we are not allowed to use names that start with underscore and then a capital letter.